src/basic/mountpoint-util.c

```diff
@@ -509,3 +512,19 @@ int mount_propagation_flags_from_string(const char *name, unsigned long *ret) {
                 return -EINVAL;
         return 0;
 }
+
+int make_mount_point_inode(const char *source, const char *dest, mode_t mode)
+{
+        struct stat st;
+
+        assert(source);
+        assert(dest);
+
+        if (stat(source, &st) < 0)
+                return -errno;
+
+        if (S_ISDIR(st.st_mode))
+                return mkdir_p_label(dest, mode);
+        else
+                return touch_file(dest, true, USEC_INFINITY, UID_INVALID, GID_INVALID, mode & ~0111);
```

Comment:
actually, "mknod()" with S_IFREG might be the better approach. it's cheaper (one syscall only) and we unlike touch_file() we don't care about the mtime of the result.

Reply:
yeah I was wondering which one to use, since both are used. I'll switch.